packages/preact/reference-app/routes/Defer.tsx

```diff
@@ -32,21 +33,28 @@ function ErrorElement() {
 }
 
 export default function Defer() {
+  let location = useLocation();
   let data = useLoaderData() as LoaderData;
 
   return (
     <>
       <p id="critical-data">Critical Data: {data.critical}</p>
 
-      <Suspense fallback={<p id="lazy-value">Loading data...</p>}>
+      <Suspense
+        fallback={<p id="lazy-value">Loading data...</p>}
+        key={`success-${location.key}`}
+      >
         <Await resolve={data.lazy}>
           {(value: Awaited<typeof data.lazy>) => (
             <p id="lazy-value">Value: {value}</p>
           )}
         </Await>
       </Suspense>
 
-      <Suspense fallback={<p id="lazy-error">Loading error...</p>}>
+      <Suspense
+        fallback={<p id="lazy-error">Loading error...</p>}
+        key={`error-${location.key}`}
+      >
```

Comment:
@jacob-ebey So this is a total hack but solves the issue for the time being 🤷‍♂️.  I made an issue in the `Projects` view to look into solving the never-resolving promise issue to unblock this